fix!: Remove `lib` from `.wxt/tsconfig.json`

yyyanghj opened the pull request:
`lib` is mismatched cause we manual set `lib` option so that override the default value which effected by `target` option.

<img width="826" alt="image" src="https://github.com/wxt-dev/wxt/assets/34713301/e8a6b599-a684-4182-912d-e6c437c77dac">

aklinker1 commented:
Hmm, I think it makes more sense to remove the `lib` defaults all together.

I added it because by default, `WebWorker` types aren't added. They are *sometimes* useful for MV3 service workers, but rarely. If we exclude the `lib` from `.wxt/tsconfig.json`, like Nuxt does, it will include the `ESNext` libs because the target is `ESNext`.

If people want web worker types, they can add them to their own tsconfig. I can add some docs around this.

yyyanghj commented:
> Thanks for the PR!
> 
> It makes more sense to remove the `lib` defaults all together. Can you update your PR?
> 
> I added it because `WebWorker` types aren't present by default. They are _sometimes_ useful for MV3 service workers, but rarely. If we exclude the `lib` from `.wxt/tsconfig.json`, like Nuxt does, it will include the `ESNext` lib by default (because the target is `ESNext`).
> 
> If people want web worker types, they can add them to their own tsconfig. I can ad docs around TSConfig customization.

OK. I have updated PR.

aklinker1 reviewed (approved):
Thanks

aklinker1 commented:
I'll fix the tests in a few hours, I'm traveling and my internet is bad. Just need to update some snapshots.

aklinker1 commented:
@yyyanghj Released in [`v0.9.0`](https://github.com/wxt-dev/wxt/releases/tag/v0.9.0)
